haystack/graph_retriever/text_to_sparql.py

```diff
@@ -0,0 +1,93 @@
+import logging
+from typing import Optional
+
+from haystack.graph_retriever.base import BaseGraphRetriever
+
+from transformers import BartForConditionalGeneration, BartTokenizer
+
+logger = logging.getLogger(__name__)
+
+
+class Text2SparqlRetriever(BaseGraphRetriever):
+    """
+        Graph retriever that uses a pre-trained Bart model to translate natural language questions given in text form to queries in SPARQL format.
+        The generated SPARQL query is executed on a knowledge graph.
+    """
+
+    def __init__(self, knowledge_graph, model_name_or_path, top_k: int = 1):
+        self.knowledge_graph = knowledge_graph
+        self.model = BartForConditionalGeneration.from_pretrained(model_name_or_path, force_bos_token_to_be_generated=True)
+        self.tok = BartTokenizer.from_pretrained(model_name_or_path)
+        self.top_k = top_k
+
+    def retrieve(self, question_text: str, top_k: Optional[int] = None):
+        if top_k is None:
+            top_k = self.top_k
+        inputs = self.tok([question_text], max_length=100, truncation=True, return_tensors='pt')
+        temp = self.model.generate(inputs['input_ids'],
+                                   num_beams=self.top_k,
```

Comment:
num beams is related to beamsearch. It should be fixed at 5-10 or with a separate parameter. Otherwise we get bad results for topk=1

Reply:
fixed it to 5 for now